src/UI/Component/Table/Factory.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/* Copyright (c) 2017 Nils Haagen <nhaageng@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\Table;
+
+/**
+ * Table factory
+ */
+interface Factory {
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *       The Presentation Table lists some data from the system in a pleasant way.
+	 *       The user should be able to get a quick overview over records in a
+	 *       dataset as simple as possible; for this, only most relevant fields of
```

Comment:
How does some field (is is not a cell if it is a table?) qualify as most relevant? can some indication be given?

Reply:
That is something the developer decides when defining the "row mapping closure". That is a device that decides how the data from a record is mapped to the row shown on the screen. Which fields are the "most relevant" should be rooted somewhere in the use case. I imagine that this could also well be switched by a mode view control. There is no automatism in the Presentation Table that decides on that.

Is there anything in this you want me to add to the docu?